result/result.py

```diff
@@ -51,20 +55,20 @@ def __repr__(self) -> str:
 
     @classmethod
     @overload
-    def Ok(cls) -> 'Result[E, bool]':
+    def Ok(cls) -> 'Result[bool, Any]':
```

Comment:
I was wondering if the `Any`s here can be swapped with `E`s and `T`s. But that's a discussion for another time.

Reply:
No, they can't :slightly_smiling_face: The reason is that at this time, only the type `T` is known, but the error type `E` is unspecified. The type can be "refined" or "narrowed" once it is being used in a context where the error type can be derived.

If we add an `E` parameter, then the user will always have to add a type annotation when creating a `Result` like that.